Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -0,0 +1,877 @@
+                    return
+                }
+            }
+        }
+        else {
+          
+            $nonCompliantMDCTierResourcetype | ForEach-Object {
+                $resource = $_
+                try {
+
+                    if ( $EnableDatabases -eq $true -and ($_.Name -eq "CosmosDbs" -or $_.Name -eq "OpenSourceRelationalDatabases" -or $_.Name -eq "SqlServers" -or $_.Name -eq "SqlServerVirtualMachines")) {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+
+                    if ($EnableResourceManager -eq $true -and $_.Name -eq "Arm") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+
+                    if ($EnableAppService -eq $true -and $_.Name -eq "AppServices") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+
+                    if ($EnableStorage -eq $true -and $_.Name -eq "StorageAccounts") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier  -SubPlan DefenderForStorageV2 
+                    }
+
+                    if ($EnableContainer -eq $true -and $_.Name -eq "Containers") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+
+                    if ($EnableServers -eq $true -and $_.Name -eq "VirtualMachines") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+
+                    if ($EnableKeyVault -eq $true -and $_.Name -eq "KeyVaults") {
+                        $remediatedResource = Set-AzSecurityPricing -Name $_.Name -PricingTier $reqMDCTier 
+                    }
+                    
+                    if (($remediatedResource | Measure-Object).Count -gt 0)
+                    {
+                        $remediatedResources += $remediatedResource | Select-Object  @{N = 'Id'; E = { $resource.Id }},
+                        @{N='Name';E={$resource.Name}},
+                        @{N='CurrentPricingTier';E={$reqMDCTier}},
+                        @{N='PreviousPricingTier';E={$resource.PricingTier}},
+                        @{N='IsPreviousProvisioningStateRegistered';E={$previousProviderRegistrationState}}
+                    }
+                    
+                }
+                catch {
+                    Write-Host "Error occurred while setting $reqMDCTier pricing tier on resource [$($_.Name)]. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                    $skippedResources += $resource | Select-Object  @{N = 'Id'; E = { $resource.Id }},
+                        @{N='Name';E={$resource.Name}},
+                        @{N='CurrentPricingTier';E={$resource.PricingTier}},
```

Comment:
both will b represented by same value?

Reply:
Yes because these resources are not remediated due to some error so previous and current pricing tier will be same